lib/features/login/presentation/login_controller.dart

```diff
@@ -53,21 +53,21 @@ class LoginController extends GetxController {
 
   var loginState = LoginState.IDLE.obs;
 
-  String _urlText = '';
-  String _userNameText = '';
-  String _passwordText = '';
+  String? _urlText;
+  String? _userNameText;
+  String? _passwordText;
 
   void setUrlText(String url) => _urlText = url.formatURLValid();
 
   void setUserNameText(String userName) => _userNameText = userName;
 
   void setPasswordText(String password) => _passwordText = password;
 
-  Uri _parseUri(String url) => Uri.parse(url);
+  Uri? _parseUri(String? url) => url != null ? Uri.parse(url) : null;
 
-  UserName _parseUserName(String userName) => UserName(userName);
+  UserName? _parseUserName(String? userName) => userName != null ? UserName(userName) : null;
 
-  Password _parsePassword(String password) => Password(password);
+  Password? _parsePassword(String? password) => password != null ? Password(password) : null;
```

Comment:
we have a ton of way to handling input. We should not transfer it to domain business.